crates/codegen/parser/runtime/src/napi_interface/cst.rs

```diff
@@ -16,6 +15,16 @@ pub enum NodeType {
     Token,
 }
 
+impl RustNode {
```

Comment:
I wonder if this is a good use case for `impl Into<Either<RuleNode, RustNode>> for RustNode` instead of the custom `into_either()`.

Reply:
Sounds good. In that case, I suggest using `into_node()` instead of `into_either()` for readability.
But up to you if you feel strongly about the current naming.

Reply:
I'm worries this doesn't follow the convention and it's not immediately clear why `into_node` will return `Either<RuleNode, TokenNode>` as it encodes additional assumption that we're talking about JS `type Node = Rule Node | TokenNode` that we add as an alias elsewhere in the templates, so hurts discoverability a bit.

I will implement `From`/`Into` for completeness but will prefer the explicit conversion in the code; I'll use `into_either_node` or `into_js_either_node` then, which should make things more immediately readable.

Reply:
I've tried to hit the middle ground and I changed the conversion to use `Into::into` in obvious cases (like functions with just `node.clone().into()`, as the return type in the signature makes it obvious) but use the explicit one when called in a `.map(...)` inside a chain to make it more obvious what's going on.